sdk/core/azure-core/test/ut/transport_policy_options.cpp

```diff
@@ -377,5 +355,484 @@ namespace Azure { namespace Core { namespace Test {
+        {
+          EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
+        }
+      }
+    }
+
+    // Now verify that once we enable CRL checks, we can still access the URLs.
+    GTEST_LOG_(INFO) << "Test with CRL checks enabled";
+    {
+      Azure::Core::Http::Policies::TransportOptions transportOptions;
+
+      // Note that the default is to *disable* CRL checks, because they are disabled
+      // by default. So we test *enabling* CRL validation checks.
+      transportOptions.EnableCertificateRevocationListCheck = true;
+      HttpPipeline pipeline(CreateHttpPipeline(transportOptions));
+
+      for (auto const& target : testUrls)
+      {
+        GTEST_LOG_(INFO) << "Test " << target;
+        Azure::Core::Url url(target);
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, url);
+        std::unique_ptr<Azure::Core::Http::RawResponse> response;
+        EXPECT_NO_THROW(
+            response = pipeline.Send(request, Azure::Core::Context::ApplicationContext));
+        if (response && response->GetStatusCode() != Azure::Core::Http::HttpStatusCode::Found)
+        {
+          EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
+        }
+      }
+    }
+
+    // Now verify that once we enable CRL checks, we can still access the URLs.
+    GTEST_LOG_(INFO) << "Test with CRL checks enabled. Iteration 2.";
+    {
+      Azure::Core::Http::Policies::TransportOptions transportOptions;
+
+      // Note that the default is to *disable* CRL checks, because they are disabled
+      // by default. So we test *enabling* CRL validation checks.
+      transportOptions.EnableCertificateRevocationListCheck = true;
+      HttpPipeline pipeline(CreateHttpPipeline(transportOptions));
+
+      for (auto const& target : testUrls)
+      {
+        GTEST_LOG_(INFO) << "Test " << target;
+        Azure::Core::Url url(target);
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, url);
+        std::unique_ptr<Azure::Core::Http::RawResponse> response;
+        EXPECT_NO_THROW(
+            response = pipeline.Send(request, Azure::Core::Context::ApplicationContext));
+        if (response && response->GetStatusCode() != Azure::Core::Http::HttpStatusCode::Found)
+        {
+          EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
+        }
+      }
+    }
+  }
+
+  TEST_F(TransportAdapterOptions, TestRootCertificate)
+  {
+    // On Windows and OSX, setting a root certificate disables the default system certificate
+    // store. That means that if we set the expected certificate, we won't be able to connect to
+    // the server because the certificates root CA is not in the store.
+#if defined(AZ_PLATFORM_LINUX)
+    // cspell:disable
+    std::string azurewebsitesCertificate
+        = "MIIF8zCCBNugAwIBAgIQCq+mxcpjxFFB6jvh98dTFzANBgkqhkiG9w0BAQwFADBh"
```

Comment:
Could these certs be retrieved from KeyVault as part of test setup?  That way we don't have the secret in the repo and its easy to rotate em if needed

Reply:
Possibly. I got these by running `openssl s_client -connect  azuresdkforcpp.azurewebsites.net:443`. 

The test proxy certificate came from eng\common\testproxy dotnet-devcert.crt.

Any rotation would have to be manually done, so I'm not 100% sure that putting the certificates in keyvault would help. And it would mean that teh azure core tests took a dependency on the keyvault SDK, which is quite challenging from a layering perspective.